nextcord/polls.py

```diff
@@ -0,0 +1,446 @@
+            "duration": self.duration,
+            "allow_multiselect": self.allow_multiselect,
+        }
+        if self.layout_type:
+            payload["layout_type"] = self.layout_type.value
+        return payload
+
+    def add_answer(self, text: str, emoji: Optional[EmojiInputType] = None) -> Self:
+        """Add a choice i.e. answer to the poll.
+
+        This function returns the class instance to allow for fluent-style chaining.
+
+        Parameters
+        ----------
+        text: :class:`str`
+            The text description of the choice.
+        emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`, :class:`str`]]
+            The emoji description of the choice.
+        """
+        self.answers.append(PollAnswer(poll_media=PollMedia(text=text, emoji=emoji)))
+        return self
+
+
+class Poll:
+    """A poll object.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    message: :class:`Message`
+        The message this poll belong to.
+    question: :class:`PollMedia`
+        The question being asked in a poll.
+    answers: List[:class:`PollAnswer`]
+        The answers of the poll.
+    expiry: :class:`datetime.datetime`
+        The expiry of the poll.
+    allow_multiselect: :class:`bool`
+        Whether this poll allow people to select more than 1 option.
+    layout_type: :class:`PollLayoutType`
+        The layout type of the poll.
+    results: Optional[:class:`PollResults`]
+        The results of the poll.
+    """
+
+    __slots__ = (
+        "message",
+        "question",
+        "_state",
+        "answers",
+        "expiry",
+        "allow_multiselect",
+        "layout_type",
+        "results",
+    )
+
+    def __init__(self, data: PollData, *, message: Message, state: ConnectionState) -> None:
+        self.message: Message = message
+        self._state: ConnectionState = state
+        self.question: PollMedia = PollMedia.from_dict(data["question"])
+        self.answers: List[PollAnswer] = [
+            PollAnswer.from_dict(poll_answer) for poll_answer in data["answers"]
+        ]
+        self.expiry: datetime = datetime.fromisoformat(data["expiry"])
+        self.allow_multiselect: bool = data["allow_multiselect"]
+        self.layout_type: PollLayoutType = try_enum(PollLayoutType, data["layout_type"])
+
+        self.results: Optional[PollResults] = (
+            PollResults(data["results"], self) if "results" in data else None
+        )
+
+    def __repr__(self) -> str:
+        return (
+            "<Poll "
+            f"message={self.message!r} "
+            f"question={self.question!r} "
+            f"answers={self.answers!r} "
+            f"expired={self.expired!r}"
+            ">"
+        )
+
+    def __eq__(self, other) -> bool:
+        return isinstance(other, self.__class__) and other.message.id == self.message.id
+
+    def __ne__(self, other) -> bool:
+        return isinstance(other, self.__class__) and other.message.id != self.message.id
```

Comment:
I'm not sure if this method is necessary at all. I don't have time to test it, but it is mentioned in the [python docs](https://docs.python.org/3/reference/datamodel.html#object.__ne__)
**tldr:** For `__ne__()`, by default it delegates to `__eq__()` and inverts the result unless it is NotImplemented.

However, if we want to leave it for clarity, it can be simplified to:
```suggestion
        return not self.__eq__(other)
```

Reply:
The result of this method is incorrect too, it returns `False` for "not equal" -> equal, if it is a different type.

Reply:
What? Returns `True` if the `equal` method returns `False`

Reply:
I am not on about your suggestion, but about the code this is on. `        return isinstance(other, self.__class__) and other.message.id == self.message.id` is correct for `__eq__` as it must be the same type and same id, but `        return isinstance(other, self.__class__) and other.message.id != self.message.id` is incorrect for `__ne__` as it returns `False` if it is not the same type, meaning `poll != user` would be `False` despite them being different things.

Reply:
That's right too, I hadn't noticed that